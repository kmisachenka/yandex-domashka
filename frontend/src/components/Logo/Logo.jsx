import React from 'react';
import * as propTypes from 'prop-types';
import { Link } from 'react-router-dom';

import styles from './Logo.module.scss';

export default function Logo(props) {
  const { service } = props;
  return (
    <div className={styles.wrapper}>
      <Link to="/">
        <img src="https://yastatic.net/q/logoaas/v1/Яндекс.svg" alt="Яндекс" />
      </Link>
      <Link to="/" className={styles.service}>
        <img
          src={`https://yastatic.net/q/logoaas/v1/${service}.svg`}
          alt={`${service}`}
        />
      </Link>
    </div>
  );
}

Logo.defaultProps = {
  service: 'Домашка',
};

Logo.propTypes = {
  service: propTypes.string,
};
