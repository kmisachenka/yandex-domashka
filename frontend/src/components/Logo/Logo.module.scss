.wrapper {
  display: flex;
}

.service {
  margin-left: 6px;
}
