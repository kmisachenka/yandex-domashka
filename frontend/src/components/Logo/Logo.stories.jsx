import React from 'react';
import { storiesOf } from '@storybook/react';
import Logo from './Logo';

storiesOf('Логотип', module)
  .add('Домашка', () => (
    <Logo />
  ))
  .add('кастомный сервис', () => (
    <Logo service="ШРИ" />
  ));
