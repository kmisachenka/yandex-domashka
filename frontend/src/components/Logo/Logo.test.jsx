import React from 'react';
import { shallow } from 'enzyme';
import Logo from './Logo';

describe('Logo', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with custom service', () => {
    const wrapper = shallow(<Logo service="Test" />);
    expect(wrapper).toMatchSnapshot();
  });
});
