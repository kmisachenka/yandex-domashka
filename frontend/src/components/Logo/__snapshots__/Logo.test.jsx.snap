// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should match snapshot 1`] = `
<div
  className="wrapper"
>
  <Link
    to="/"
  >
    <img
      alt="Яндекс"
      src="https://yastatic.net/q/logoaas/v1/Яндекс.svg"
    />
  </Link>
  <Link
    className="service"
    to="/"
  >
    <img
      alt="Домашка"
      src="https://yastatic.net/q/logoaas/v1/Домашка.svg"
    />
  </Link>
</div>
`;

exports[`Logo should render with custom service 1`] = `
<div
  className="wrapper"
>
  <Link
    to="/"
  >
    <img
      alt="Яндекс"
      src="https://yastatic.net/q/logoaas/v1/Яндекс.svg"
    />
  </Link>
  <Link
    className="service"
    to="/"
  >
    <img
      alt="Test"
      src="https://yastatic.net/q/logoaas/v1/Test.svg"
    />
  </Link>
</div>
`;
